File: String/第一个只出现一次的字符.py
# -*- coding:utf-8 -*-
class Solution:
    def FirstNotRepeatingChar(self, s):
        # write code here
        from collections import defaultdict
        hashTable = defaultdict(int)
        for i in s:
            hashTable[i] += 1
        for j in hashTable:
            if hashTable[j] == 1:
                return s.find(j)
        return 0
        
               

File: String/第一个只出现一次的字符.cpp
class Solution {
public:
    int FirstNotRepeatingChar(string str) {
        if(str.size() == 0)
            return -1;
        int occurence[256];
        for(int i=0; i<256; i++){
            occurence[i] = -1;
        }
        for(int i=0; i<str.size(); i++){
            if(occurence[str[i]] == -1){
                occurence[str[i]] = i;
            }else if(occurence[str[i]] >=0){
                occurence[str[i]] = -2;
            }
        }
        int index = 10000;
        for(int i=0; i<256; ++i){
            if(occurence[i] >= 0 && index > occurence[i])
                index  = occurence[i];
        }
        if(index >= 0)
            return index;
        else
            return -1;
    }
};